core/src/main/java/org/apache/carbondata/core/scan/model/QueryModel.java

```diff
@@ -124,6 +124,11 @@
 
   private boolean preFetchData = true;
 
+  /**
+   * It fills the vector directly from decoded column page with out any staging and conversions
```

Comment:
"It fills the vector", can you give more detail for which vector? and describe how spark/presto is integrated with this?

Reply:
ok, added comment